### PR TITLE
Use name of parameter if text is not available

### PR DESCRIPTION
**Pull request by kozaxinan**
Fixes https://github.com/slackhq/compose-lints/issues/273

For standard compose functions, `method.uastParameters` returns parameter with delegated to KtUltraLightParameterForSource. KtUltraLightParameterForSource has proper text function to return.

When method is generic with reified, delegated uast becomes UastKotlinPsiParameter and getText of UastKotlinPsiParameter returns null.

Those are from my investigation.

In this PR I suggest checking name of the parameter if text fails. If you know better way/ correct way to get text when function is reified, I can update the solution.

Example reified usage is `androidx.lifecycle.viewmodel.compose.viewModel`. We had a wrapper with wrong parameter order to hit this issue.

### CODE REVIEW

**Review by ZacSweers — APPROVED** [review]
Nice find and investigation, thanks!

File: compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt [hunk 1]
```diff
@@ -120,6 +131,10 @@ class ParameterOrderDetectorTest : BaseComposeLintTest() {
           @@ -17 +17
           - fun MyComposable(text1: String, m2: Modifier = Modifier, modifier: Modifier = Modifier, trailing: () -> Unit) { }
           + fun MyComposable(text1: String, modifier: Modifier = Modifier, m2: Modifier = Modifier, trailing: () -> Unit) { }
+          Fix for src/test.kt line 20: Replace with (modifier, text, lambda):
+          @@ -20 +20
+          - inline fun <reified T> MyComposable(text: String = "123", modifier: Modifier = Modifier, lambda: () -> Unit) : T { }
+          + inline fun <reified T> MyComposable(modifier, text, lambda) : T { }
```
Comment (kozaxinan) [inline compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt:137]: I actually didn't like this part. I will check if I can make it better.

File: compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt [hunk 2]
```diff
@@ -120,7 +131,11 @@ class ParameterOrderDetectorTest : BaseComposeLintTest() {
           @@ -17 +17
           - fun MyComposable(text1: String, m2: Modifier = Modifier, modifier: Modifier = Modifier, trailing: () -> Unit) { }
           + fun MyComposable(text1: String, modifier: Modifier = Modifier, m2: Modifier = Modifier, trailing: () -> Unit) { }
-          """
+          Fix for src/test.kt line 20: Replace with (modifier: Modifier, text: String, lambda: Function0<Unit>):
+          @@ -20 +20
+          - inline fun <reified T> MyComposable(text: String = "123", modifier: Modifier = Modifier, lambda: () -> Unit) : T { }
+          + inline fun <reified T> MyComposable(modifier: Modifier, text: String, lambda: Function0<Unit>) : T { }
```
Comment (ZacSweers) [inline compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt:137]: Can we unpack lambdas?

Comment (kozaxinan) [reply compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt:137]: Let me check. I don't know how to get lambda when text is not available. I will check about sorting that differently.

Comment (ZacSweers) [reply compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt:137]: You should be able to just get the text range for that PSI element in the worst case, which would let you just get the `() ->` function syntax. My thinking is that this is the kind of thing that could be just a rearrangement of existing text, and you could do that replace in a stringy way

Comment (kozaxinan) [reply compose-lint-checks/src/test/java/slack/lint/compose/ParameterOrderDetectorTest.kt:137]: Thank you for PSI element idea. I updated code to create correct error message and lint fix.

**Review by ZacSweers — APPROVED** [review]
Nice!

**Pull request merged**